src/Camelotia.Presentation.Avalonia/Views/ProviderView.xaml

```diff
@@ -17,25 +19,128 @@
+                         FontSize="14"
+                         Margin="15 0" />
+                <ListBox Grid.Column="1"
+                         IsVisible="{Binding ShowBreadCrumbs}"
+                         Items="{Binding BreadCrumbs}"
+                         ScrollViewer.VerticalScrollBarVisibility="Hidden"
+                         ScrollViewer.HorizontalScrollBarVisibility="Hidden"
+                         Margin="5">
+                  <ListBox.Styles>
+                    <Style Selector="ListBox > Border">
+                      <Setter Property="CornerRadius" Value="5"/>
+                      <Setter Property="Background" Value="{DynamicResource ThemeCardBrush}"/>
+                    </Style>
+                    <Style Selector="Button">
+                      <Setter Property="Margin" Value="0"/>
+                      <Setter Property="Padding" Value="5,0,0,0"/>
+                      <Setter Property="VerticalContentAlignment" Value="Center"/>
+                      <Setter Property="Background" Value="Transparent"/>
+                    </Style>
+                    <Style Selector="Menu">
+                      <Setter Property="Padding" Value="0"/>                      
+                      <Setter Property="Background" Value="Transparent"/>                      
+                      <Setter Property="Foreground" Value="{DynamicResource ThemeForegroundBrush}"/>
+                    </Style>
+                    <Style Selector="MenuItem">
+                      <Setter Property="Padding" Value="0"/>
+                      <Setter Property="Margin" Value="2,0"/>                      
+                    </Style>
+                    <Style Selector="MenuItem:pointerover /template/ Border#root">
+                      <Setter Property="Background" Value="{DynamicResource HighlightBrush}" />
+                    </Style>
+                    <Style Selector="MenuItem /template/ Border#root">
+                      <Setter Property="CornerRadius" Value="5" />
+                    </Style>
+                    <Style Selector="ListBoxItem">
+                      <Setter Property="Padding" Value="0"/>
+                      <Setter Property="Margin" Value="0"/>
+                      <Setter Property="BorderThickness" Value="0"/>
+                    </Style>                
+                    <Style Selector="TextBlock">                      
+                      <Setter Property="VerticalAlignment" Value="Center"/>                      
+                    </Style>
+                  </ListBox.Styles>
+                  <ListBox.ItemsPanel>
+                    <ItemsPanelTemplate>
+                      <StackPanel Orientation="Horizontal"/>
+                    </ItemsPanelTemplate>
+                  </ListBox.ItemsPanel>
+                  <ListBox.ItemTemplate>
+                    <DataTemplate>
+                      <Button Command="{Binding Provider.SetPath}" CommandParameter="{Binding FullPath}">
```

Comment:
Probably worth moving this data template to a separate file. The pattern used in the project currently maps a view model class to a view class, so assuming we introduced a new `FolderViewModel`, worth introducing a new user control implementing `ReactiveUserControl<FolderViewModel>` and named `FolderView`.

Also, this way we could incorporate [Pharmacist](https://github.com/reactiveui/pharmacist) and use `this.topLevelMenu.Events()` instead of a `ChangePropertyAction` https://www.reactiveui.net/docs/handbook/events/ A couple of reasons for this. The `.Events()` pattern is [already used in the project](https://github.com/worldbeater/Camelotia/blob/main/src/Camelotia.Presentation.Avalonia/Views/FileView.xaml.cs#L21), and ReactiveUI docs generally [recommend](https://www.reactiveui.net/docs/handbook/events/#prefer-reactiveui.events-over-xaml-behaviors) using observables instead of events and behaviors (`.Events()` are strongly typed, and C# is more expressive compared to XML). E.g. we could keep both drawing presenters and toggle their `IsVisible` properties based on `IsSubMenuOpen` value or an event like 'opening' or 'closing'.

Reply:
Very good suggestions